File: MixAndMatch/MixAndMatch/Body.cpp
#include "Body.h"

Body::Body()
{
	addParts();
}

Body::~Body()
{
}

pair<string, gcroot<Bitmap^>> Body::cratePair(string s, gcroot<Bitmap^> b)
{
	return make_pair(s, b);
}

void Body::addParts()
{
	dictionary.insert(cratePair("fairy", createBitmap("Fairy_1.png")));
	dictionary.insert(cratePair("frankenstein", createBitmap("Frankenstein_1.png")));
	dictionary.insert(cratePair("skeleton", createBitmap("Skeleton_1.png")));
	dictionary.insert(cratePair("vampire", createBitmap("Vampire_1.png")));
	dictionary.insert(cratePair("werewolf", createBitmap("Werewolf_1.png")));
	dictionary.insert(cratePair("witch", createBitmap("Witch_1.png")));
}

gcroot<Bitmap^> Body::createBitmap(string file)
{
	gcroot<Bitmap^> bitmap;
	bitmap->FromFile("C:/Users/´÷Àû·á/Documents/Visual Studio 2015/Projects/MixAndMatch/Pictures/" + gcnew String(file.c_str()));
	return bitmap;
}

File: MinAndMatchCSharp/MinAndMatchCSharp/Body.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MinAndMatchCSharp
{
    class Body : BodyParts
    {
        public Body()
        {
            addParts();
        }

        protected override void addParts()
        {
            dictionary.Add("fairy", createBitmap("Fairy_1.png"));
            dictionary.Add("frankenstein", createBitmap("Frankenstein_1.png"));
            dictionary.Add("skeleton", createBitmap("Skeleton_1.png"));
            dictionary.Add("vampire", createBitmap("Vampire_1.png"));
            dictionary.Add("werewolf", createBitmap("Werewolf_1.png"));
            dictionary.Add("witch", createBitmap("Witch_1.png"));
        }
    }
}
